pkg/gofr/datasource/pubsub/google/google.go

```diff
@@ -117,63 +122,71 @@ func (g *googleClient) Publish(ctx context.Context, topic string, message []byte
 }
 
 func (g *googleClient) Subscribe(ctx context.Context, topic string) (*pubsub.Message, error) {
-	receiveChan := make(chan *pubsub.Message)
-	defer close(receiveChan)
+	var end time.Duration
 
-	ctx, span := otel.GetTracerProvider().Tracer("gofr").Start(ctx, "gcp-subscribe")
+	spanCtx, span := otel.GetTracerProvider().Tracer("gofr").Start(ctx, "gcp-subscribe")
 	defer span.End()
 
-	g.metrics.IncrementCounter(ctx, "app_pubsub_subscribe_total_count", "topic", topic, "subscription_name", g.Config.SubscriptionName)
+	g.metrics.IncrementCounter(spanCtx, "app_pubsub_subscribe_total_count", "topic", topic, "subscription_name", g.Config.SubscriptionName)
 
-	t, err := g.getTopic(ctx, topic)
-	if err != nil {
-		return nil, err
-	}
+	if _, ok := g.subStarted[topic]; !ok {
```

Comment:
This if is covering so many lines.

I would have moved the code currently inside to an unexported method